oceannavigator/frontend/src/components/MapToolbar.jsx

```diff
@@ -74,36 +76,71 @@ class MapToolbar extends React.Component {
     this.props.action(name);
   }
   
-  class4ButtonHandler() {
-    const button = $(ReactDOM.findDOMNode(this.class4button));
-    if (this.class4Picker && this.class4Picker.is(":visible")) {
-      this.class4Picker.hide();
-    } else if (!this.class4Picker) {
-      if (this.class4Picker !== null) {
-        this.class4Picker = null;
+  class4OpButtonHandler() {
+    const button = $(ReactDOM.findDOMNode(this.class4OpButton));
+    if (this.class4OpPicker && this.class4OpPicker.is(":visible")) {
+      this.class4OpPicker.hide();
+    } else if (!this.class4OpPicker) {
+      if (this.class4OpPicker !== null) {
+        this.class4OpPicker = null;
       }
-      this.class4Picker = $(this.class4div).datepicker({
+      this.class4OpPicker = $(this.class4OpDiv).datepicker({
         dateFormat: "yy-mm-dd",
-        beforeShowDay: this.beforeShowDay.bind(this),
+        beforeShowDay: this.beforeShowDay_op.bind(this),
         regional: this.props.i18n.language,
         onSelect: function(text, picker) {
-          this.props.action("show", "class4", this.state.class4Files[text]);
-          this.class4Picker.hide();
+          this.props.action("show", "class4", this.state.class4Files_op[text], "class4_op");
+          this.class4OpPicker.hide();
         }.bind(this),
         defaultDate: this.state.class4Current,
       });
```

Comment:
Trying to find a way to simplify this section so that we only need one buttonhandler method. I'd like to do something like this but I'm having trouble finding away to get references to `this.class4OpDiv` and `this.class4RoaDiv` working here. Any suggestions @htmlboss @prodyut1978 ?
```suggestion
class4ButtonHandler(class4type) {
    const button = null;
    const div = null;
    const files = null;
    switch (class4type) {
      case "class4_op":
        button = $(ReactDOM.findDOMNode(this.class4OpButton));
        div = this.class4OpDiv;  // doesn't work below
        files = this.state.class4Files_op[text];
        break;
      case "class4_rao":
        button = $(ReactDOM.findDOMNode(this.class4RoaButton));
        div = this.class4RaoDiv;
        files = this.state.class4Files_rao[text];
        break;
    }
    if (this.class4Picker && this.class4Picker.is(":visible")) {
      this.class4Picker.hide();
    } else if (!this.class4Picker) {
      if (this.class4Picker !== null) {
        this.class4Picker = null;
      }
      this.class4Picker = $(div).datepicker({
        dateFormat: "yy-mm-dd",
        beforeShowDay: this.beforeShowDay_op.bind(this),
        regional: this.props.i18n.language,
        onSelect: function(text, picker) {
          this.props.action("show", "class4", files, class4type);
          this.class4Picker.hide();
        }.bind(this),
        defaultDate: this.state.class4Current,
      });

      $(div).css("left", button.offset().left + "px");
    } else {
      this.class4Picker.show();
    }
    this.forceUpdate();
  }
```
```